Sources/AmazonLocationiOSAuthSDK/Auth/AuthHelper.swift

```diff
@@ -22,29 +24,40 @@ import AwsCommonRuntimeKit
     private func authenticateWithCognitoIdentityPoolAndRegion(identityPoolId: String, region: String) async throws -> LocationCredentialsProvider? {
         let credentialProvider = LocationCredentialsProvider(region: region, identityPoolId: identityPoolId)
         credentialProvider.setRegion(region: region)
-        try await credentialProvider.getCognitoProvider()?.refreshCognitoCredentialsIfExpired()
+        amazonLocationClient = AmazonLocationClient(locationCredentialsProvider: credentialProvider)
+        try await amazonLocationClient?.initialiseLocationClient()
         return credentialProvider
     }
 
-    @objc public func authenticateWithApiKey(apiKey: String, region: String) -> LocationCredentialsProvider {
+    @objc public func authenticateWithApiKey(apiKey: String, region: String) async throws -> LocationCredentialsProvider? {
         let credentialProvider = LocationCredentialsProvider(region: region, apiKey: apiKey)
         credentialProvider.setAPIKey(apiKey: apiKey)
         credentialProvider.setRegion(region: region)
-        locationCredentialsProvider =  credentialProvider
+        locationCredentialsProvider = credentialProvider
+        if let credentialsProvider = locationCredentialsProvider {
+            amazonLocationClient = AmazonLocationClient(locationCredentialsProvider: credentialsProvider)
+            try await amazonLocationClient?.initialiseLocationClient()
+        }
         return credentialProvider
     }
     
     public func authenticateWithCredentialsProvider(credentialsProvider: CredentialsProvider, region: String) async throws -> LocationCredentialsProvider? {
         let credentialProvider = LocationCredentialsProvider(credentialsProvider: credentialsProvider)
         credentialProvider.setRegion(region: region)
+        locationCredentialsProvider = credentialProvider
+        if let credentialsProvider = locationCredentialsProvider {
+            amazonLocationClient = AmazonLocationClient(locationCredentialsProvider: credentialsProvider)
+        }
         return credentialProvider
     }
     
-    @objc public func getLocationClient() -> AmazonLocationClient?
+    @objc public func getAmazonLocationClient() -> AmazonLocationClient?
     {
-        guard let locationCredentialsProvider = self.locationCredentialsProvider else {
-            return nil
-        }
-        return AmazonLocationClient(locationCredentialsProvider: locationCredentialsProvider)
+        return amazonLocationClient
+    }
+    
+    public func getLocationClient() -> LocationClient?
+    {
+        return amazonLocationClient?.locationClient
```

Comment:
This is changing the public API so that getLocationClient() now returns LocationClient instead of AmazonLocationClient. It seems like this would break any existing code. Why is this change necessary?

Reply:
(For example see here - https://github.com/aws-geospatial/amazon-location-samples-ios/blob/main/quick-start/QuickStartTrackingApp/ViewModels/TrackingViewModel.swift#L85 )

Looking deeper into the auth helper, I think I understand why you want to expose the underlying location client, since I'm not sure how you would use AmazonLocationClient without extending it. But I think we need to do this in a way that wouldn't break any existing code.